### PR TITLE
Move the device ID generation to ratbagctl

### LINKED ISSUE #641: Drop the DBus Device.Id
Devices are separate objects on DBus and the device ID is an opaque string. So anyone who needs a device ID can just use the object path or make up their own ID. No need for ratbagd to do this.

### PR DESCRIPTION
**Pull request by whot**
See #641 for a longer explanation.

Fixes #641

### CODE REVIEW

**Comment by whot** [comment]
Urgh. this fails the test because our ratbagctl test suite crazy merge mix doesn't fetch the custom device names in the fast run (and the slow run fails elsewhere).

**Comment by whot** [comment]
Alright, updated and ready for review now :)

File: tools/ratbagctl.in [hunk 1]
```diff
@@ -1504,10 +1504,48 @@ def get_parser():
...
+            'screaming', 'yelling', 'yodeling', 'singing',
+            'honking', 'hooting', 'whispering', 'hollering',
+    ]
+
+    # Let's convert the sha into something not boring. This takes the first
+    # 4 characters, creates two different indices from it to generate a
+    # name. The rest is hope hope that never get a collision here but it's
+    # unlikely enough.
+    name = device_names[int(device.id[0], 16) + int(device.id[1], 16)]
+    attr = device_attr[int(device.id[2], 16) + int(device.id[3], 16)]
```
Comment (bentiss) [inline tools/ratbagctl.in]: Hmm, this is not a proper random distribution. Here you are basically saying: "we take 2 dice (of 16), we roll them and get the sum of it to fetch the index". And that means you basically have much less chances of picking index `0` than `16` as `0 = 0 + 0` but `16 = 0 + 16 = 1 + 15 = 2 + 14 = ...` 

Assuming the device.id is at least 8 chars, we could do:
```
name = device_names[(int(device.id[0:2], 16) + int(device.id[2:4], 16)) % len(device_names)]
attr = device_attr[(int(device.id[4:6], 16) + int(device.id[6:8], 16)) % len(device_attr)]
```

This gives makes the distribution more flat as we are summing 2 dices that have a bigger size than each array.

Note that with that, in the first, third, fifth and seventh chars only the first bit will be taken into account (but does it really matters?)

File: tools/ratbagctl.in [hunk 2]
```diff
@@ -1466,10 +1466,48 @@ def get_parser():
...
+            'yodeling', 'singing', 'honking', 'hooting',
+            'whispering', 'hollering',
+    ]
+
+    # Let's convert the sha into something not boring. This takes the first
+    # 4 characters, creates two different indices from it to generate a
+    # name. The rest is hope hope that never get a collision here but it's
+    # unlikely enough.
+    name = device_names[int(device.id[0:2], 16) % len(device_names)]
+    attr = device_attr[int(device.id[2:4], 16) % len(device_attr)]
```
Comment (bentiss) [inline tools/ratbagctl.in]: Thanks for changing this. But now the commit message is wrong, can you update it?

File: tools/ratbagctl.in (hunk 1 above)
Comment (whot) [reply tools/ratbagctl.in]: Fixed this by taking the first 4 bytes as 2-byte hex values and running a `% len(...)` on it. Also removed some of the entries I didn't like.

File: tools/ratbagctl.in (hunk 2 above)
Comment (whot) [reply tools/ratbagctl.in]: done!

**Pull request merged**